artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyAcceptor.java

```diff
@@ -863,17 +863,19 @@ public NettyAcceptor setShutdownTimeout(int shutdownTimeout) {
    }
 
    private static String getProtocols(Map<String, ProtocolManager> protocolManager) {
-      StringBuilder sb = new StringBuilder();
-      if (protocolManager != null) {
+      if (protocolManager == null) {
```

Comment:
how many times this is supposed to happen?

Reply:
I don't know, but code like this just hurts my eyes.

Reply:
I would actually prefer a early exit using `if(protocolManager == null) return "";` but that would result in more lines being touched.

Reply:
For me it is also about reducing the cognitive load when reading source code. Now you know directly when a null was specified you get an empty string.

Reply:
it's fine as long is more readable...
> but code like this just hurts my eyes.

Agree, but I would focus on things that hurt/benefit common usage, not just our eyes :P 
Although I agree is nice to clean up stuff, to make it more readable :)

Reply:
@franz1981 I just came across it while trying to understand/analyze the whole class to better understand what may cause https://issues.apache.org/jira/browse/ARTEMIS-3117. Maybe you can have a look there too. :-D

Reply:
taking a look although i suggest to run async-profiler on it, disabling Dyantrace; that would help finding native hotspots too (if SSL is involved on Netty side).